map_view/display-device-location-with-autopan-modes/src/main/java/com/esri/samples/display_device_location_with_autopan_modes/DisplayDeviceLocationWithAutopanModesSample.java

```diff
@@ -0,0 +1,200 @@
+        "/display_device_location_with_autopan_modes/polyline_data.json"), StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline locations = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWgs84());
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        locations, new SimulationParameters(Calendar.getInstance(), 5.0, 0.0, 0.0));
+
+      // configure the map view's location display to follow the simulated location data source
+      LocationDisplay locationDisplay = mapView.getLocationDisplay();
+      locationDisplay.setLocationDataSource(simulatedLocationDataSource);
+      locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      locationDisplay.setInitialZoomScale(1000);
+
+      // toggle the location display visibility on check
+      checkBox.setOnAction(event -> {
+        if (checkBox.isSelected()) {
+          // start the location display
+          locationDisplay.startAsync();
+
+        } else {
+          // turn off the location display
+          locationDisplay.stop();
+        }
+        // toggle the combo box interactions
+        comboBox.setDisable(!checkBox.isSelected());
+      });
+
+      // set the autopan mode of the location display based on the mode chosen from the combo box
+      comboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+
+        // set the scale that the map view will zoom to when the autopan mode is changed
+        locationDisplay.setInitialZoomScale(1000);
+
+        switch (comboBox.getSelectionModel().getSelectedItem()) {
+          case "Off":
+            locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.OFF);
+            break;
+          case "Recenter":
+            locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+            break;
+          case "Navigation":
+            locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.NAVIGATION);
+            break;
+          case "Compass":
+            locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.COMPASS_NAVIGATION);
+            break;
+        }
+      });
+
+      // enable the checkbox and combo box interactions when the map is loaded
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          checkBox.setDisable(false);
+          checkBox.setSelected(true);
+          comboBox.setDisable(false);
+
+          // start the location display
+          locationDisplay.startAsync();
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load: " + map.getLoadError().getCause().getMessage()).show();
+        }
+      });
```

Comment:
since this is what happens right after loading the sample, it makes sense to move this just after the locationDisplay initializer steps (and before the checkBox on action events).